plumbing/transport/http/common.go

```diff
@@ -673,20 +673,22 @@ func checkError(r *http.Response) error {
 		}
 	}
 
+	err := &Err{
+		URL:    r.Request.URL,
+		Status: r.StatusCode,
+		Reason: reason,
+	}
+
 	switch r.StatusCode {
 	case http.StatusUnauthorized:
-		return transport.ErrAuthenticationRequired
+		return fmt.Errorf("%s %w", transport.ErrAuthenticationRequired.Error(), err)
 	case http.StatusForbidden:
-		return transport.ErrAuthorizationFailed
+		return fmt.Errorf("%s %w", transport.ErrAuthorizationFailed.Error(), err)
 	case http.StatusNotFound:
-		return transport.ErrRepositoryNotFound
+		return fmt.Errorf("%s %w", transport.ErrRepositoryNotFound.Error(), err)
```

Comment:
We should wrap both errors here `fmt.Errorf("%w: %w", ...)`, otherwise, using `errors.Is(...)` would fail on transport errors.